apps/prairielearn/src/sync/fromDisk/sharing.sql

```diff
@@ -0,0 +1,46 @@
+-- BLOCK sync_question_sharing_sets
+WITH
+  nqss AS (
+    SELECT
+      *
+    FROM
+      jsonb_to_recordset($new_question_sharing_sets::JSONB) AS (question_id bigint, sharing_set_id bigint)
+  )
+INSERT INTO
+  sharing_set_questions (question_id, sharing_set_id)
+SELECT
+  question_id,
+  sharing_set_id
+FROM
+  nqss
+ON CONFLICT (question_id, sharing_set_id) DO NOTHING;
+
+--BLOCK sync_course_sharing_sets
+WITH
+  ncss AS (
+    SELECT
+      *
+    FROM
+      jsonb_to_recordset($new_course_sharing_sets::JSONB) AS (name text, description text)
+  )
+INSERT INTO
+  sharing_sets (course_id, name, description)
+SELECT
+  $course_id,
+  name,
+  description
+FROM
+  ncss
+ON CONFLICT (course_id, name) DO
+UPDATE
+SET
+  description = EXCLUDED.description;
```

Comment:
is there a way to make this prettier? I struggle to get JSON into SQL queries without being extremely verbose.

Reply:
You should be able to use `jsonb_to_recordset(...)` directly in the `FROM` clause here, in place of `ncss`.

Reply:
got it working, thanks!